File: library/src/main/java/com/getstream/sdk/chat/rest/request/ReactionRequest.java
package com.getstream.sdk.chat.rest.request;

import com.getstream.sdk.chat.model.Reaction;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.HashMap;
import java.util.Map;

public class ReactionRequest {

    @SerializedName("reaction")
    @Expose
    private
    Map<String, Object> data;

    public ReactionRequest(Reaction reaction) {
        HashMap<String, Object> data;
        if (reaction.getExtraData() != null) {
            data = new HashMap<>(reaction.getExtraData());
        } else {
            data = new HashMap<>();
        }
        data.put("type", reaction.getType());
        this.data = data;
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/rest/ReactionRequest.kt
package io.getstream.chat.android.core.poc.library.rest

import com.google.gson.annotations.SerializedName
import io.getstream.chat.android.core.poc.library.Reaction


class ReactionRequest(reaction: Reaction) {

    @SerializedName("reaction")
    val data: Map<String, Any>

    init {
        this.data = reaction.extraData.toMutableMap()
        this.data["type"] = reaction.type
    }
}